tools/test-js-with-node

```diff
@@ -155,7 +173,21 @@ print('Starting node tests...')
 
 # If we got this far, we can run the tests!
 try:
-    ret = subprocess.check_call(command)
+    # This is duplicate code of:
+    #   `def run_parallel(lint_functions)` via commands.py
+    # A possible TODO: Refactor the function to work for both cases.
+    pids = []
+    for tests in sub_tests:
+        pid = os.fork()
+        if pid == 0:
+            os.execvp('node', command + tests)
```

Comment:
I think something is fishy over here. If we think about it, after the first fork there would be still stuff left in the sub_tests, right. And basically the child process will continue the loop from there only hence forking more childs which isn't really something we want the child to be doing.
I am gonna test my theory by some print statements but I do feel its probably right

Reply:
yup verified, this will cause issues of repetitive creation of child for same set of tests.

Reply:
Sorry for the churn, everything seems fine. I just commented out the wrong line and got confused

Reply:
Okay, I see what you're saying.  So my understanding that led to this was as follows:

I thought `os.execvp` would take the child process and mutate it into a node process who would never return to the parent, such that the only way to monitor status would be through referencing the pid.  Therefore a child wouldn't return to the loop as it becomes a different process whose code has been replaced.

I understand I might be looking at this wrong, and am interested in what your testing shows.  I certainly expected a little better performance, and this theory makes sense as to why that isn't showing.

Reply:
Yeah you are right and the behaviour you described is exactly what's happening. I actually was also trying to figure out why didn't we see a much better performance boost and than formulated the theory and basically when testing it out I made a mistake and got confused.
Sorry for that.

Reply:
No problem! :)